### ISSUE TITLE
Source Control badge doesn't appear

### ISSUE THREAD

**Issue by BrenoVelasco** [getcursor/cursor#771]
The badge indicating the number of files changed in Source Control icon doesn't appear. It does appear, however, in the Changes dropdown. Is there a way to make it visible? Thanks.

<img width="444" alt="Screen Shot 2023-08-22 at 14 18 55" src="https://github.com/getcursor/cursor/assets/21113340/de37f9c0-27bf-435d-bd58-0363118e3b3e">

**Comment by atul-jalan** [comment]
+1 Please fix this! I rely on this a decent amount to quickly see how many changes I've made.

**Comment by CHATGPTGOAT** [comment]
I am actually troubleshooting this issue now but first a quick explanation
of how this happened.
1. On my first visit to your amazing site when I was prompted to create the
account somehow I simply entered in my email and verified.
2. After watching more youtube videos on how awesome you guys are I
decided to head back to your site and signup.
3. It was during this enrollment visit that I subconsciously enrolled with
my Google account - paid the upgrade fee and started building my prompt for
CTRL K.
4. It was when I returned to upload my prompt that I discovered I screwed
up bigtime by crossing that login process - as a developer I should have
known better.

5. From one developer to another I first would like to say I'm sorry for
generating a trouble ticket that should have never been because my entire
onboarding was flawless. So what is the best way to sort this out - please
advise?

Having said that, I am here because I believe in all of the potential and
possibilities of Large Language Models. and When I stumbled upon
Cursor I became an instant fan. Please help me sort out the mess I made so
I can join you guys in changing the world.

**Comment by ansh** [comment]
I'm having this issue too. Is there any way to turn on source control badge?

**Comment by ansh** [comment]
Temp fix: Change Activity Bar Orientation to vertical

![CleanShot 2024-06-11 at 15 41 20@2x](https://github.com/getcursor/cursor/assets/8207733/4af8d98c-3dc3-44d8-8307-a7619ff8e096)

**Comment by AXSJ** [comment]
Thanks @ansh but I prefer the horizontal orientation? Ideal solution would be to keep it as horizontal but show the badge

**Comment by theshaneyu** [comment]
+1 this is extremely important to me

**Comment by ggous** [comment]
I can confirm this.
But the solution @ansh posted works!

**Comment by lyallh** [comment]
> Temp fix: Change Activity Bar Orientation to vertical
> 
> ![CleanShot 2024-06-11 at 15 41 20@2x](https://private-user-images.githubusercontent.com/8207733/338763539-4af8d98c-3dc3-44d8-8307-a7619ff8e096.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjUwNzk0NTAsIm5iZiI6MTcyNTA3OTE1MCwicGF0aCI6Ii84MjA3NzMzLzMzODc2MzUzOS00YWY4ZDk4Yy0zZGMzLTQ0ZDgtODMwNy1hNzYxOWZmOGUwOTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgzMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MzFUMDQzOTEwWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODVhZDY5NGEzZjZlMDdkY2I5ODhhNDY4MGZiZDc5OGQ2OTcxODgwMTFkMDAwZDRkM2M4NDQxMDIwODY3N2RlMCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.bsXrXzLXxPCC9bIwZGz3iTU7o-3QMTnwUC-2xJboq5g)

Also change the Activity Bar: **Location** to top @AXSJ @theshaneyu 

![image](https://github.com/user-attachments/assets/bd35fe7e-da61-4b1c-8c48-7a4bad17bdc2)

Looks like Orientation really switches between a custom Cursor activity bar and Vscode's built in one. The vscode one has a horizontal/top position we can use for a working badge.

**Comment by daawascript** [comment]
That works but I do prefer Cursor's custom activity bar, looking forward to getting this fixed and I might upgrade to pro when it is.

**Comment by shadeemerhi** [comment]
Also hopeful for this fix!

**Comment by 0mjs** [comment]
tldr; use the non-custom original orientation where it isn't _broken_

thanks!

**Comment by ViktorJJF** [comment]
Hi, im new in cursor and I also want this fix on the horizontal orientation plsssssss

**Comment by FerminBD** [comment]
Any news on this?

**Comment by Hextar** [comment]
I've just found a nice solution to show the **source control pending changes badge**, while still sparing some pixels while using my laptop.

![image](https://github.com/user-attachments/assets/78db7ed1-88df-4b1e-a24c-1bea86e0873a)

**Comment by theshaneyu** [comment]
@Hextar 's settings work for me. Thank you so much

**Comment by sebukaimp** [comment]
Badges should be on by default.  It is extremely important.  As of December 2024 the issue is still happening.  I used the vertical workspace workaround.  A dealbreaker for me, really.

**Comment by julee** [comment]
The latest version still has this issue. Is it unsolvable? It's been a long time.

**Comment by Liyumk** [comment]
This bothers me a lot, the badges should be on by default

**Comment by theolvq** [comment]
The latest version actually broke @Hextar's solution. Now we cannot use VS Code's default activity bar, if we do, the badges get cut off: 

![Image](https://github.com/user-attachments/assets/95c18eab-dcf0-4746-b128-e7e96cda2d5a)

**Comment by andrewduss** [comment]
Super annoying that it doesn't show by default.

Changing the activity bar location to `default` and orientation to `vertical` will show the badges again after the March 2025 update (Unsure of the exact date)

![Image](https://github.com/user-attachments/assets/115e3159-3732-453c-8725-c2e05553ca05)

**Comment by ChipLuxury-EWA** [comment]
24/3/2025 - same bug:

![Image](https://github.com/user-attachments/assets/3b91cc40-57f3-47d4-b4be-da199d3d7c82)

the fix suggesting to switch to vsc og activity bar is not good

![Image](https://github.com/user-attachments/assets/73ffe76f-cad9-47da-999b-6aa21909853d)

is ms having patent in activity bar??? why not just using the og one?

**Comment by ZhenyaGoroh** [comment]
In a horizontal mode the badge doesn't exist, in a vertical mode it is a half-cut

<img width="1318" alt="Image" src="https://github.com/user-attachments/assets/11524f33-5473-4665-8453-aeb768001583" />

**Comment by WhiskeyBest** [comment]
Why is this STILL a problem, after all this time??

The badge **only** shows in vertical mode, and **NOT** on the default horizontal one. 

Is this only affecting a small number of people?

**Comment by ViktorJJF** [comment]
The bug still exists, but as a workaround you can add this to your settings.json for horizontal mode

![Image](https://github.com/user-attachments/assets/73a76814-e04b-42dc-b487-cbfc1aa1645f)

**Comment by KovalchukKonstantin** [comment]
Having the same issue on critical workflow

**Comment by enriquechoza** [comment]
I was very annoyed by this so I went to update the source code. I changed the `top` property from `17px` to `9px` in the css rule `.monaco-workbench .pane-composite-part>.header-or-footer>.composite-bar-container>.composite-bar>.monaco-action-bar .action-item.icon .badge.compact .badge-content`

You can find the file in `/Applications/Cursor.app/Contents/Resources/app/out/vs/workbench/workbench.desktop.main.css`

Maybe it's going to be overriden in the next update, if that's the case I might consider writing a small script to run after updating.

**Before**

<img width="384" alt="Image" src="https://github.com/user-attachments/assets/76bb6dd4-d069-475b-a3a4-534e39e7faab" />

**After**

<img width="384" alt="Image" src="https://github.com/user-attachments/assets/1b68ad51-f8ea-47b3-b1e3-1271b68eeadc" />

**Comment by germy-BD** [comment]
![Image](https://github.com/user-attachments/assets/91743273-31ff-4b95-897a-ea46a1e9cc03)

thanks @enriquechoza! this actually works. you can just copy and paste this code: `.monaco-workbench .pane-composite-part>.header-or-footer>.composite-bar-container>.composite-bar>.monaco-action-bar .action-item.icon .badge.compact .badge-content` and then cursor will do the changes (it requires `sudo` privileges), after that you need to restart Cursor and the changes appear. 

Happy coding!

**Comment by ChipLuxury-EWA** [comment]
@enriquechoza  THANKS!!

here is the file location on windows: 
```
"C:\Users\yourUserName\AppData\Local\Programs\cursor\resources\app\out\vs\workbench\workbench.desktop.main.css"
```
search in the file for: 
```
.monaco-workbench .pane-composite-part>.header-or-footer>.composite-bar-container>.composite-bar>.monaco-action-bar .action-item.icon .badge.compact .badge-content,.monaco-workbench .pane-composite-part>.title>.composite-bar-container>.composite-bar>.monaco-action-bar .action-item.icon .badge.compact .badge-content{border-radius:16px;font-size:9px;font-weight:600;height:13px;line-height:13px;min-width:13px;padding:0 2px;position:absolute;right:0;text-align:center;top:17px}
```

and change the 17 to 9:

![Image](https://github.com/user-attachments/assets/3a97024a-812c-4cca-b1ec-fd1717350484)

Restart cursor

----

Since this fix is taking some screen-estate, for me is a temporary fix and I hope to see this bug fix in cursor-activity-bar:
BEFORE:
![Image](https://github.com/user-attachments/assets/706165ee-3a28-4259-818f-c7bbfa762ca5)
AFTER:
![Image](https://github.com/user-attachments/assets/b0eebf0c-2fb9-4dbe-84ec-707543be0280)

**Comment by bryanprimus** [comment]
bug still exist, i'm on mac

**Comment by aslamdoctor** [comment]
I ended up installing **Custom UI Style** extension and added below configurations to `settings.json`

```
"workbench.activityBar.orientation": "vertical",
"workbench.activityBar.location": "top",
"custom-ui-style.stylesheet": {
        ".monaco-workbench .pane-composite-part>.header-or-footer>.composite-bar-container>.composite-bar>.monaco-action-bar .action-item.icon .badge.compact .badge-content": "top: 9px",
        ".monaco-workbench .composite-bar .monaco-action-bar .action-item .action-label:before": "font-size:16px !important"
},
```

This is how it looks now :)

<img width="289" alt="Image" src="https://github.com/user-attachments/assets/990f6832-2154-4bee-a3e1-a41be2ce922a" />

**Comment by rogerhlg** [comment]
the bug is still a bug...

**Comment by ravshansbox** [comment]
It is not a "can we" issue, it is a "do we care" issue

**Comment by dbsxdbsx** [comment]
> Fixing this will improve the QoL drastically...

Exactly. But, unfortunately, it is still not fixed at the latest version of 0.20.

**Comment by pray3m** [comment]
+1 , why is this still not fixed ?
